pkg/oci/oci.go

```diff
@@ -871,6 +876,17 @@ func ensureImage(ctx context.Context, imageStore oras.Target, image string, imgO
 		}
 	}
 
+	if imgOpts.Registry != "" {
+		targetImage, err := normalizeImageName(image)
```

Comment:
Not related to this PR in particular, but: we're using the `normalizeImageName` _a LOT_. Do we ever use the non-normalized name? If not, does it make sense to do the normalization at the highest level already and always pass the normalized name down?